### PR TITLE
Hourly rent

### PR DESCRIPTION
**Pull request by jonzee**

### CODE REVIEW

File: src/ts/FormVG/index.spec.tsx [hunk 1]
```diff
@@ -26,18 +26,25 @@ describe('<FormVG />', () => {
...
-  it('check for form inner components', () => {
+  it('check for form inner components in day form', () => {
     expect(wrapper.find('FormWrap')).to.have.length(1);
-    expect(wrapper.find('FormPure')).to.have.length(2);
+    expect(wrapper.find('FormDay')).to.have.length(1);
     expect(wrapper.find('Contact')).to.have.length(1);
   });
 
-  it('check if simple methods works fine', () => {
+  it('check for form inner components in hour form', () => {
```
Comment (michalmikolajczyk) [inline src/ts/FormVG/index.spec.tsx]: please clarify

File: src/ts/FormVG/index.spec.tsx [hunk 2]
```diff
@@ -26,18 +26,25 @@ describe('<FormVG />', () => {
...
+    wrapper = shallow(<FormVG offer={offer} hour={true} />, formsyContext());
+    expect(wrapper.find('FormWrap')).to.have.length(1);
+    expect(wrapper.find('FormHour')).to.have.length(1);
+    expect(wrapper.find('Contact')).to.have.length(1);
+  });
+
+  it('check if displaying title & price works properly', () => {
```
Comment (michalmikolajczyk) [inline src/ts/FormVG/index.spec.tsx]: ```it('should display the title and price', ...```

File: src/ts/FormVG/index.spec.tsx [hunk 3]
```diff
@@ -26,18 +26,25 @@ describe('<FormVG />', () => {
...
     expect(instance.getTitle(instance.props.offer)).to.be.equal(offer.title);
     expect(instance.getPrice(instance.props.offer)).to.be.equal(renderUnit(instance.props.offer));
   });
 
-  it('check if total calculation works fine', () => {
+  it('check if total calculation for day form works fine', () => {
```
Comment (michalmikolajczyk) [inline src/ts/FormVG/index.spec.tsx]: ```it('should ... for the ... component', ...```

File: src/ts/helpers/hours.ts [hunk 4]
```diff
@@ -8,16 +8,20 @@ const subtypes = [
   'off-road',
 ];
 
+export function isHour(offer): boolean {
```
Comment (michalmikolajczyk) [inline src/ts/helpers/hours.ts]: please clarify what this does? Does it check if the passed argument is an hour? Or does it check if the offer can be booked by the hour? I believe the latter, but it better be represented in the function name.

File: src/ts/helpers/hours.ts [hunk 5]
```diff
@@ -8,16 +8,20 @@ const subtypes = [
...
+  if (!offer) return false;
+  return _.includes(subtypes, offer.subtype);
```
Comment (michalmikolajczyk) [inline src/ts/helpers/hours.ts]: this is not specific for hourly rentals. something like this perhaps? `_.get(offer, 'subtypes.hourly')`

**Comment by lam3lka** [comment]
Według mnie wszystko OK - śmiga!

File: src/ts/helpers/hours.ts (hunk 5 above)
Comment (jonzee) [inline src/ts/helpers/hours.ts]: For now we don't have in offer model props that determines if offer is hourly or daily, we just check if offer subtype is in `[groupon, dual-sport, off-road]`, and if so we show hourly form. IMO we should add `prices` prop in api with `hour`, `day`, `week` and `month` arrays with prices and allow user to choose in `base price` what rental plan he wants.

**Pull request merged**